lib/simplecov/coverage_statistics.rb

```diff
@@ -35,22 +35,22 @@ def initialize(covered:, missed:, total_strength: 0.0)
       @covered  = covered
       @missed   = missed
       @total    = covered + missed
-      @percent  = compute_percent(covered, total)
-      @strength = compute_strength(total_strength, @total)
+      @percent  = compute_percent(covered)
+      @strength = compute_strength(total_strength)
     end
 
   private
 
-    def compute_percent(covered, total)
-      return 100.0 if total.zero?
+    def compute_percent(covered)
+      return 100.0 if @missed.zero?
 
-      covered * 100.0 / total
+      covered * 100.0 / @total
     end
 
-    def compute_strength(total_strength, total)
-      return 0.0 if total.zero?
+    def compute_strength(total_strength)
+      return 0.0 if @total.zero?
```

Comment:
The usage of parameters here (and rest of the file) was intentional. That way it's clearer to me what part of the computation relies on which values just by looking at the method signature/seeing their calls in initialize :)

I know one could debate it isn't very OO and if they were public it wouldn't be like this, but personally I like them better like this :)